LoopKit/GlucoseEffect.swift

```diff
@@ -26,3 +26,30 @@ extension GlucoseEffect: Comparable {
         return lhs.startDate < rhs.startDate
     }
 }
+
+public extension Array where Element == GlucoseEffect {
+    func asVelocities() -> [GlucoseEffectVelocity] {
+        guard count > 1 else {
+            return []
+        }
+        let unit = HKUnit.milligramsPerDeciliter
+        var previousEffectValue: Double = first!.quantity.doubleValue(for: unit)
+        var previousEffectDate: Date = first!.startDate
+
+        var velocities = [GlucoseEffectVelocity]()
+
+        for effect in self {
+            let value = effect.quantity.doubleValue(for: unit)
+            let delta = value - previousEffectValue
+            let timespan = effect.startDate.timeIntervalSince(previousEffectDate).minutes
+            let velocity = delta / timespan
```

Comment:
wouldn't the first pass used the same element to determine `previousEffectValue` and `previousEffectDate`? If yes, then wouldn't this cause a divide by 0 (assuming `timespan` would be 0).

Reply:
Yes, good catch! This was producing an extra element in the output, with NaN as a value.  Will post a fix, thanks.